actionpack/lib/action_dispatch/routing/inspector.rb

```diff
@@ -86,12 +85,15 @@ def format(formatter, filter = nil)
 
       private
 
-      def filter_routes(filter)
-        if filter
-          @routes.select { |route| route.defaults[:controller] == filter }
+      def filter_routes(options)
+        routes = if options[:controller]
+          filter = options[:controller].downcase.sub(/_?controller\z/, '')
+          @routes.select { |r| r.defaults[:controller] == filter }
         else
           @routes
         end
+
+        routes.select { |r| /#{options[:pattern]}/ =~ r.defaults.values.join(' ') }
```

Comment:
Usually I'm used to seeing the pattern on the right hand side, why are you doing it the other way around? :smile: